File: go/grains/grains.go
package grains

import "errors"

const numberOfSquares = 64

// Square calculates the number of grains on the given square.
func Square(n int) (uint64, error) {
	if n < 1 || n > numberOfSquares {
		return 0, errors.New("n is out of bounds")
	}

	return 1 << (n - 1), nil
}

// Total calculates the total number of grains on a chess board.
func Total() uint64 {
	var grains uint64 = 0
	for i := 1; i <= numberOfSquares; i++ {
		n, _ := Square(i)
		grains += n
	}
	return grains
}

File: swift/grains/Sources/Grains/Grains.swift
enum GrainsError: Error {
  case inputTooLow
  case inputTooHigh
}

private let size = 64

struct Grains {
  static func square(_ num: Int) throws -> UInt64 {
    guard num > 0 else { throw GrainsError.inputTooLow }
    guard num <= size else { throw GrainsError.inputTooHigh }
    return 1 << (num - 1)
  }

  static var total: UInt64 {
    get throws {
      UInt64.max
    }
  }
}
